### ISSUE TITLE
mandelbrot generated by wbf2c encounters SegFault

### ISSUE THREAD

**Issue by jserv** [skeeto/bfc#1]
While executing the compiled C program generated by `wbf2c` on [mandelbrot.bf](https://github.com/ErikDubbelboer/brainfuck-jit/blob/master/mandelbrot.bf), it soon encounters segmentation fault.

The way to reproduce:
- ./wbf2c mandelbrot.b  > m.c
- gcc -o m m.c && ./m

Tested on Linux/x86_64.

**Comment by skeeto** [comment]
Yeah, that's definitely a bug. I wrote this 8 years ago, back when I was still a college undergrad, so there are probably lots and lots of bugs like this (which will never be fixed). It doesn't have a README.md because the latest commit pre-dates GitHub itself. How did you come across this?

If you're interested in a project like this, I highly recommend my much more recent (May 2015) bf-x86 instead: https://github.com/skeeto/bf-x86. No intermediate C, it compiles directly to optimized x86_64 machine code, so there are no dependencies on an external compiler and it can run programs JIT-style. It's far more interesting.

**Comment by jserv** [comment]
@skeeto , Great! bf-x86 is superior, and I like its functionality to generate ELF directly.

**Comment by jserv** [comment]
By the way, is it possible to apply the multithreaded code generation techniques into [bf-x86](https://github.com/skeeto/bf-x86)?

**Comment by skeeto** [comment]
It's not currently a feature and would take some work to add it, but it
_could_ be done without the hassle of introducing linkage:

http://nullprogram.com/blog/2015/05/15/

I'm not convinced there's much value in it, though. It was just a
experiment I tried on this old project.
